### ISSUE TITLE
Lasagna (basics) feedback

### ISSUE THREAD

**Issue by angelikatyborska** [exercism/go#1745]
Hi! I'm going through the very first learning exercise on the track as a student who has never written a single line of Golang before. I managed to get through it on my own without cheating, so that's great :) but I still have some suggestions for improvements.

Below might be a lot text, but the kind of changes that I am suggesting would be actually tiny, like adding an extra word here and there, or a single sentence.

## 1. Functions: multiple return values?
> Functions may have multiple (explicitly typed) return values. 

This sounds mindblowing to me. How can a single function have multiple return values? I have never seen this before. Or am I misinterpreting things and does this mean only that the type definition of the return value allows something like a union type? Anyway, my problem here is that there is no follow up to that statement. No examples of a function with "multiple return values", whatever that means in practice. Is it even relevant to the exercise? If yes, maybe a code example would be nice to have. If not, maybe it doesn't need to say anything? I have no idea. I was just confused by this sentence.

## 2. Lowercase names vs uppercase names

In the example code, a variable's name is lowercase while a constant's name is uppercase. Is this a requirement? Is it just a convention? Is it just like with function names, that both variable names and constant names can be either lowercase or uppercase and that changes whether they are public or not? I wish there was at least one sentence for each explaining the name casing, like there is for functions.

## 3. Variable names - camelCase or not?

I don't see any hints in the introduction or in the code itself as to whether I'm supposed to give variable names using camelCase or snake_case, or something else. I do have examples that constants and functions are named in PascalCase, and a variable name is supposed to be lowercase, so probably camelCase, but a specific mention of this would be nice for a total Golang beginner.

## 4. Types? I don't know them... 

So, I need to define the functions from zero, there are no function stubs. I don't know any types in Golang beyond `string` and `int` that are in the code examples in the introduction.

When I defined the function without arguments as a start, I got this compilation error:

```
# lasagna [lasagna.test]
./lasagna_test.go:39:31: too many arguments in call to RemainingOvenTime
	have (number)
	want ()
./lasagna_test.go:39:31: RemainingOvenTime(25) used as value
./lasagna_test.go:63:29: too many arguments in call to PreparationTime
	have (int)
	want ()
./lasagna_test.go:63:29: PreparationTime(tt.layers) used as value
./lasagna_test.go:88:25: too many arguments in call to ElapsedTime
	have (int, int)
	want ()
```

From this compilation error, I was convinced that `number` is the correct type of the argument in `RemainingOvenTime`, but that was wrong. They're all supposed to be `int`s. Could the instructions maybe say that? "...function that takes the actual minutes the lasagna has been in the oven (`int`)"? 

Why is this compilation error saying `have (number)` and not `have (int)` for `RemainingOvenTime`?

---

That's all my thoughts! I hope it's not too much. I can probably answer a lot of those questions myself by googling, but I wanted to share this because I think a lot of Golang beginners might have similar doubts.

**Comment by bobtfish** [comment]
This is great feedback.

Re `Functions: multiple return values` I think we should just remove this from the lasagna exercise and docs once PR#1720 is merged, as they're covered in much more detail (and with examples) in the functions concept + they're not needed for the basic lasagna exercise.

The other things you called out should definitely be addressed in this concept / example IMO.

I'll be happy to take care of this all once PR#1720 is merged if a maintainer wants to assign this to me.

**Comment by junedev** [comment]
@bobtfish Totally agree regarding multiple return values.

It would be great if you could give lasagna some cleanup.

I just had a longer discussion with @SaschaMann https://exercism-team.slack.com/archives/CC4UPN5KP/p1633203810038400.
We think there are two general things that would help to make lasagna and other "early" exercises better/easier for students:
1. Have proper stub file like e.g. this one https://github.com/exercism/go/blob/main/exercises/concept/cars-assemble/cars_assemble.go. That would eliminate people getting stuck on issue like this https://github.com/exercism/go/issues/1763 and not being able to run the tests at all for a long time (until they worked out all the correct function signatures).
2. Include examples of common compiler errors in the hints or if they are concept related, then in the introduction. E.g. in lasagna the hint for the first task could include what you see if you don't yet have the constant defined. So the hint (either for the task or general) could be something along the lines `If you see `undefined: xyz` when trying to run the test, double check that ...`
